mappings/net/minecraft/structure/pool/StructurePoolBasedGenerator.mapping

```diff
@@ -1,7 +1,29 @@
 CLASS net/minecraft/class_3778 net/minecraft/structure/pool/StructurePoolBasedGenerator
 	FIELD field_16665 LOGGER Lorg/slf4j/Logger;
 	METHOD method_27230 generate (Lnet/minecraft/class_7138;IZLnet/minecraft/class_2794;Lnet/minecraft/class_3485;Lnet/minecraft/class_5539;Lnet/minecraft/class_5819;Lnet/minecraft/class_2378;Lnet/minecraft/class_3790;Ljava/util/List;Lnet/minecraft/class_265;)V
+		ARG 0 noiseConfig
+		ARG 1 maxSize
+		ARG 2 modifyBoundingBox
+		ARG 3 chunkGenerator
+		ARG 4 structureManager
+		ARG 5 heightLimitView
+		ARG 6 random
+		ARG 7 structurePoolRegistry
+		ARG 8 lastPiece
```

Comment:
![image](https://user-images.githubusercontent.com/2484721/167285990-a2a4d35a-5f23-4d6e-b1f3-51593ac061ab.png)
Maybe I'm misunderstanding, but it looks like it's specifically added last to `structurePieces`, and then pieces are generated from that beginning with the first, so this would be the last piece, no?

Reply:
The `generate` method does BFS. `structurePieces` is a fresh empty queue when `lastPiece` is added at the tail, so `lastPiece` is the first piece (`Last` in `addLast` doesn't matter since it's an empty queue). `generatePiece` adds child pieces to `structurePieces` at the tail, so `lastPiece` is not the last piece.